Lombiq.Tests.UI/Exceptions/HtmlValidationAssertionException.cs

```diff
@@ -30,5 +35,17 @@ public HtmlValidationAssertionException(string message, Exception innerException
             : base(message, innerException)
         {
         }
+
+        private static string CreateErrorMessage(UITestContext context, bool createReportOnFailure)
+        {
+            var url = context.Driver.Url;
+            var title = context.Driver.Title ?? url;
+
+            var message = $"Asserting the HTML validation result on page {url}({title}) failed.";
+
+            return createReportOnFailure
+                ? message + " Check the HTML validation report in the failure dump for details."
+                : message;
+        }
```

Comment:
This is quite cool but should be handled more generically at a higher level, so you get this contextual information for any exception that's thrown, including axe violations (which are asserted during page changes too) or else.